### DISCUSSION TITLE
Axis size changed by label?

### DISCUSSION THREAD

**Discussion by parangpo** [oxyplot/oxyplot#2100] [Q&A] [answered]
<img width="173" alt="oxy_shrink" src="https://github.com/user-attachments/assets/94830669-4da0-4772-ba24-6d68fab8fdd2">
<img width="175" alt="oxy_expand" src="https://github.com/user-attachments/assets/cb5a3138-2af4-42e7-bc3b-3dab33f5c105">

Video: https://github.com/user-attachments/assets/5a5cafb9-6c00-4799-97f1-c6c97a868fa4

OxyPlot seems to have a bug where the length of the axis is shortened or expanded by the labels.
I am using TimeSpanAxis for the horizontal axis and LinearAxis for the vertical axis. As you look at the picture above, you can see that the horizontal axis is changed by the labels.
As far as I remember, this problem did not exist until version 2.0.

**Comment by VisualMelon** [comment, marked as answer]
That looks like expected behaviour to me (OxyPlot is designed to maximise use of space without clipping anything and doesn't cope well with artificial constraints: this can lead to weird and annoying behaviours if the data change). Looks like you have pan and zoom disabled?

In your situation, I would manually set the `PlotMarkgins` explicitly and reduce the `IntervalLength` (screen space dimensions) to ensure there is always a label on screen (or use the new `MinimumMajorIntervalCount`); this will however, lead to cropped and so potentially misleading labels on the right-hand edge if a generous enough margin is no chosen.

```
plotModel.PlotMargins = new OxyThickness(double.NaN, double.NaN, 50, double.NaN);
xAxis.IntervalLength = 50;
```

Certainly the behaviour could be better here. An option to force the pan/zoom behaviour (which leads to more stable, but not entirely predictable margins) might be useful.

For interactive plots, I usually slap a fixed and wide margin on them; there's no real alternative if you need total stability. A pair of `MinimumPlotMargins` and `MaximumPlotMargins` properties to clamp the axis margins would be useful and easy to implement.

**Reply by parangpo** [reply]
Thanks for your detail information.
I tried the method of adding a margin on the right side as you said, and it seems much better than I expected.
And, as you said also, I am not using panning. It has to show the graph for a very very long time, I am using a "circular queue" with fixed memory to save memory.
Anyway, I decided to use the method of adding a margin on the right. The right side is wasted, but I think it is an easy and simple method. Thanks alot.

**Reply by VisualMelon** [reply]
If you want to keep the space on the right, can always make the margin smaller and accept that the labels will clip; a feature to not print clipping labels would probably be a good idea as well.
